### ISSUE TITLE
`sel_is` command

### ISSUE THREAD

**Issue by kbjarkefur** [lsms-worldbank/selector#9]
@arthur-shaw , when I looked at this with fresh eyes I thought of a way that I think will be better to do the `sel_is_numeric`, `sel_is_string` commands. Creating one command for each `sel_is_???` will be a pain to maintain across that many files. So how about we make the type a sub-command. Like this: `sel_is numeric`, `sel_is string` etc. Then there is only one command `sel_is` and it accepts the sub-commands below:

- `sel_is numeric`. All numeric questions.
- `sel_is string`. All questions that contain text (i.e., Text and List types)
- `sel_is text`. Text questions only
- `sel_is list`. List question only
- `sel_is multi_select`. All multi-select questions.
- `sel_is multi_ordered`. Multi-select questions where answer order is captured.
- `sel_is multi_yn`. Multi-select with yes/no answers.
- `sel_is multi_checkbox`. Multi-select with checkbox input.
- `sel_is gps`.

I can create this quite quickly. But let me know what you think first

**Comment by kbjarkefur** [comment]
add option of `varlist(varlist)` to make it possible to pipe this from chained calls of commands in this package

**Comment by kbjarkefur** [comment]
> is_single_select. Single-select question whose answer options aren't linked to a roster ID or to a list question: type == "SingleQuestion" & (mi(linked_to_roster_id) & mi(linked_to_question_id))

I do not find the variable `linked_to_question_id` in the meta data dta file. (I do find `linked_to_roster_id`)

**Comment by kbjarkefur** [comment]
> - `is_multi_yn`. Multi-select yes/no: `type == "MultyOptionsQuestion" & yes_no_view == 1`
> - `is_multi_checkbox`. Multi-select checkbox: `type == "MultyOptionsQuestion" & yes_no_view == 1`

These seems to have a copy paste issue as they are duplicates

**Comment by kbjarkefur** [comment]
> - `is_date`. `type == "DateTimeQuestion" & is_timestamp == 0`
> - `is_timestamp`. `type == "DateTimeQuestion" & is_timestamp == 0`

I assume `is_timestamp` should be `type == "DateTimeQuestion" & is_timestamp == 1`

**Comment by arthur-shaw** [comment]
> > is_single_select. Single-select question whose answer options aren't linked to a roster ID or to a list question: type == "SingleQuestion" & (mi(linked_to_roster_id) & mi(linked_to_question_id))
> 
> I do not find the variable `linked_to_question_id` in the meta data dta file. (I do find `linked_to_roster_id`)

You're right. I updated the data files on OneDrive to include this variable. This omission is due to an error in the susometa package. This exercise is helping me improve that package. Sorry that you're suffering through that process. Thanks for flagging issues.

> > * `is_multi_yn`. Multi-select yes/no: `type == "MultyOptionsQuestion" & yes_no_view == 1`
> > * `is_multi_checkbox`. Multi-select checkbox: `type == "MultyOptionsQuestion" & yes_no_view == 1`
> 
> These seems to have a copy paste issue as the conditions are duplicates

Correct. It should have been:

* `is_multi_yn`. Multi-select yes/no: `type == "MultyOptionsQuestion" & yes_no_view == 1`
* `is_multi_checkbox`. Multi-select checkbox: `type == "MultyOptionsQuestion" & yes_no_view == 0`



> > * `is_date`. `type == "DateTimeQuestion" & is_timestamp == 0`
> > * `is_timestamp`. `type == "DateTimeQuestion" & is_timestamp == 0`
> 
> I assume `is_timestamp` should be `type == "DateTimeQuestion" & is_timestamp == 1`
> 
> EDIT: also, the value for the variable `is_timestamp` is not `0`/`1` in the meta data, it is `TRUE`/`FALSE`. I would prefer to change this to `0`/`1` in the meta data as that is Stata practice.

Your assumption is right. Sorry for the copy-paste problem.

As for the values in the metadata, I've updated the data on OneDrive to have 0/1 values. While I expected TRUE/FALSE to get automatically converted to 1/0 values, I've not added some code to convert them explicitly. For the moment, I'm simply targetting variables matching `is_*`. If there are others, please let me know.

**Comment by kbjarkefur** [comment]
Adding a reminder here for you to update the helpfile of `sel_vars` following the updates in #10
